aiida/engine/daemon/client.py

```diff
@@ -41,6 +48,23 @@
 __all__ = ('DaemonClient',)
 
 
+@dataclasses.dataclass
+class DaemonStatus:
```

Comment:
I agree that `error` is little informative and `is_reachable` is already a lot better. I am not sure about renaming `is_running` to `has_pid_file` though. The reason is that that attribute is set _not_ based on the existence of a PID file, but actually through a call of the client (if the daemon is not running, the client will return `status: 'daemon-error-not-running'`). I was actually thinking of moving away from defining the daemon as running as soon as a PID file exists, because in many cases this is simply not true and leads to inconsistencies. I think it is better to actually try to reach the daemon. So the `is_daemon_running` property should also change implementation. I deliberatly didn't change that behavior in this PR, but maybe we should just go ahead and do it.

How about the following then for the dataclass:
```python
@dataclasses.dataclass
class DaemonStatus:
    """Class to describe the status of the daemon.

    Only if both ``is_running`` and ``is_reachable`` are True`` can the daemon be assumed to be running correctly.
    """

    is_running: bool
    """Set to ``True`` if the daemon PID file exists, ``False`` otherwise."""

    is_reachable: bool
    """Set to ``False`` if the daemon process could not be reached, indicating a problem, ``True`` otherwise."""

    message: str
    """Human-readable description of the daemon status."""
```

Reply:
Sounds good to me! (just also adapt the docstring of `is_running` then)

Reply:
I had another good think and found that I didn't like this solution in the end. I realized that the `DaemonClient` interface was still not very user friendly when using it from the Python API and I think that should be the goal. The `DaemonClient` should be feature complete, and the `verdi daemon` commands are then very thin wrappers that just format the output in the style of `verdi`.

I ended up refactoring the code quite a lot so I decided to open up a new PR #5940 . That still includes the fix in the logic of the stale PID file cleaning. I admit that the changes have become significant and address multiple (usability) issues. If you think I should split it up in separate commits/PRs, please let me know and I will see what I can do. If you agree with the direction of the changes in the new PR, it can supersede this one.